### PR TITLE
doc: nrf: nrf7002: Add working with constrained host

### PR DESCRIPTION
**Pull request by krish2718**
This document provides recommendations and guidelines on how the companion chip nrf7002 should be work with hosts that are resource constrained.

@rlubos I was searching for some existing documentation on how to fine tune networking stack but nothing found at https://docs.zephyrproject.org/latest/connectivity/networking/index.html, so, I wrote a quick section, but let me know if you have any existing material I can link to.

P.S. Would really appreciate if all can push commits (we can all squash them before merge) instead of leaving commits (I just accept and squash them) except when in doubt, then please leave a suggestion for others to comment, saves everyone's time.

### CODE REVIEW

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (mjd90) [inline doc/nrf/ug_nrf7002_constrained.rst:51]:
```suggestion
NRF700X driver performance and memory fine tuning controls
```

Comment (mjd90) [inline doc/nrf/ug_nrf7002_constrained.rst:51]:
```suggestion
NRF700X driver performance and memory fine tuning controls
```

**Review by mjd90 — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (sachinthegreen) [inline doc/nrf/ug_nrf7002_constrained.rst:3]:
```suggestion
Operating with a resource constrained host
```

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (divipillai) [inline doc/nrf/ug_nrf7002_constrained.rst:14]: Please provide an introductory statement for Zephyr OS factors. The section cannot be left blank without any content.

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (divipillai) [inline doc/nrf/ug_nrf7002_constrained.rst:60]: Please confirm if the :term:`Station mode (STA)` is linking to the correct term in the glossary.
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/glossary.html#term-Station-mode-STA

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (richabp) [inline doc/nrf/ug_nrf7002_constrained.rst:48]:
```suggestion
nRF700X driver performance and memory fine-tuning knobs
```

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (richabp) [inline doc/nrf/ug_nrf7002_constrained.rst:51]:
```suggestion
The nRF700x driver provides the following software configurations to fine-tune memory and performance based on the use case:
```

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (richabp) [inline doc/nrf/ug_nrf7002_constrained.rst:56]:
```suggestion
| :kconfig:option:`CONFIG_WPA_SUPP`        | ``y`` or ``n``               | Enable or disable Wi-Fi Protected Access (WPA) supplicant                         | Memory savings                         | This specifies the inclusion of the WPA supplicant module.                                                       |
```

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (richabp) [inline doc/nrf/ug_nrf7002_constrained.rst:84]:
```suggestion
The configuration options must be used in conjunction with the Zephyr networking stack configuration options to achieve the desired performance and memory usage.
These option form a staged pipeline all the way to the nRF7002 chip, any change in one stage of the pipeline will impact the performance and memory usage of the next stage.
For example, solving bottleneck in one stage of the pipeline might lead to a bottleneck in the next stage.
```

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (richabp) [inline doc/nrf/ug_nrf7002_constrained.rst:97]:
```suggestion
     - Use cases
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/ug_nrf7002_constrained.rst
Comment (krish2718) [reply doc/nrf/ug_nrf7002_constrained.rst:60]: Yes, this is correct.

**Review by sachinthegreen — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Review by mjd90 — APPROVED** [review]
LGTM

**Review by mjd90 — CHANGES_REQUESTED** [review]
I want this document to appear underneath "Developing with nRF70 Series", at the same level as "Operating in power save modes"

**Comment by divipillai** [comment]
> > I want this document to appear underneath "Developing with nRF70 Series", at the same level as "Operating in power save modes"
> 
> Do you mean as part of `ug_nrf70_developing` ? if yes, then we need to wait till that gets merged, as that is adding this new RST file.

@richabp and I too had a discussion about the placement of this document. We agreed that once this PR is merged she would move this document under "Developing with nRF70 Series" in her PR.

**Comment by richabp** [comment]
Agree with Chaitanya. We can wait for Power Save PR to get merged.

**Comment by krish2718** [comment]
@mjd90 if you are okay with above discussions please approve.

**Review by richabp — APPROVED** [review]

**Review by mjd90 — APPROVED** [review]

**Pull request merged**
